gcode/src/parser.rs

```diff
@@ -255,11 +265,11 @@ where
             }
         }
 
-        if line.is_empty() {
-            None
-        } else {
-            Some(line)
-        }
+        // TODO: This should exit the parser under some conditions.
+        // IS M2 or M30: see 3.6.1.
+        // return None;
+
+        return Some(line);
```

Comment:
```suggestion
        Some(line)
```

Reply:
I'm going to accept this change because I'm new at Rust and after some quick research I think you have this right. But frankly, this is a weird language design decision that I don't understand. Not having a `return` statement makes it really hard to find all the exit points.